SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -1028,7 +1027,7 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
             }
         }
     } else {
-        if (body.length == 0 && attachmentIds.count < 1) {
+        if (body.length == 0 && attachmentIds.count < 1 && !contact) {
```

Comment:
Good catch!

We need something similar in OWSRecordTranscriptJob.m

```
    if (outgoingMessage.body.length < 1 && outgoingMessage.attachmentIds.count < 1) {
        OWSFail(@"Ignoring message transcript for empty message.");
        return;
    }
```

Reply:
Done.